src/common/src/types/interval.rs

```diff
@@ -65,22 +66,96 @@ impl Interval {
         }
     }
 
-    pub fn get_days(&self) -> i32 {
-        self.days
+    /// Returns the total number of whole months.
+    ///
+    /// Note the difference between [`months`] and [`months_field`].
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "5 yrs 1 month".parse().unwrap();
+    /// assert_eq!(interval.months(), 61);
+    /// assert_eq!(interval.months_field(), 1);
+    /// ```
+    pub fn months(&self) -> i32 {
+        self.months
     }
 
-    pub fn get_months(&self) -> i32 {
-        self.months
+    /// Returns the number of days.
+    pub fn days(&self) -> i32 {
+        self.days
     }
 
-    pub fn get_usecs(&self) -> i64 {
+    /// Returns the total number of microseconds in a day.
+    pub fn usecs(&self) -> i64 {
         self.usecs
     }
 
-    pub fn get_usecs_of_day(&self) -> u64 {
+    /// Calculates the remaining number of microseconds in a day.
+    ///
+    /// Note the difference between [`usecs`] and [`usecs_of_day`].
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "-1:00:00".parse().unwrap();
+    /// assert_eq!(interval.usecs(), -1 * 60 * 60 * 1_000_000);
+    /// assert_eq!(interval.usecs_of_day(), 23 * 60 * 60 * 1_000_000);
+    /// ```
+    pub fn usecs_of_day(&self) -> u64 {
         self.usecs.rem_euclid(USECS_PER_DAY) as u64
     }
 
+    /// Returns the years field.
+    pub fn years_field(&self) -> i32 {
+        self.months / 12
+    }
+
+    /// Returns the months field. range: 0-11
+    pub fn months_field(&self) -> i32 {
+        self.months % 12
+    }
+
+    /// Returns the days field.
+    pub fn days_field(&self) -> i32 {
+        self.days
+    }
+
+    /// Returns the hours field. range: -23..=23
+    pub fn hours_field(&self) -> i32 {
+        (self.usecs / USECS_PER_SEC / 3600 % 24) as i32
+    }
+
+    /// Returns the minutes field. range: -59..=-59
+    pub fn minutes_field(&self) -> i32 {
+        (self.usecs / USECS_PER_SEC / 60 % 60) as i32
+    }
+
+    /// Returns the seconds field, including fractional parts, in microseconds.
+    /// range: -59,999,999..=59,999,999
+    pub fn seconds_in_micros(&self) -> i32 {
+        (self.usecs % (USECS_PER_SEC * 60)) as i32
+    }
+
+    /// Returns the microseconds since 1970-01-01 00:00:00+00.
+    pub fn epoch_in_micros(&self) -> i128 {
+        // https://github.com/postgres/postgres/blob/REL_15_2/src/backend/utils/adt/timestamp.c#L5304
```

Comment:
Just sharing a wired behavior of PostgreSQL. Not suggesting any actions...
https://www.db-fiddle.com/f/5LLhMM7MY2DpiQwNzxYMkV/0